### DISCUSSION TITLE
Change trigger branch without a code change

### DISCUSSION THREAD

**Discussion by abel-rm-rf** [community/community#26645] [Actions]
<p><strong>Context</strong><br>
We try to use git flow.<br>
During our sprint we work on the develop branch and towards the end of the sprint we cut to release branch and make any last min changes into release branch. Meanwhile the rest of the the team keeps merging their PRs into develop. The GH action workflow watches the specified branch and deploys the code when we have a merge (into develop for example).</p>
<p><strong>Problem</strong><br>
We want to change our trigger branch when it comes towards the end of the sprint to be triggered from release branch. We can do this by changing the branch specified in the yaml file but feels like an overkill. We would ideally like to do it from somehwere else like the workflow_dispatch GUI or using a label or something that requires minimal effort for the team.</p>
<p>Here is our stripped down yaml:</p>
<pre><code class="lang-auto">name: Build &amp; Deploy
on:
  workflow_dispatch:
  push:
    branches:
      - develop # I want to change this dynamically
jobs:
  build:
    runs-on: ubuntu-latest
    - name: Install &amp; build
      run: |
        echo "Deploy code..."
</code></pre>

**Comment by jsoref** [comment]
<p>You can do what you want using:</p><aside class="onebox allowlistedgeneric" data-onebox-src="https://github.blog/changelog/2021-11-10-github-actions-input-types-for-manual-workflows/">
  <header class="source">
      <img src="https://user-images.githubusercontent.com/2119212/181099046-367894d3-15bc-4af9-b17a-613f84f70ba0.png" class="site-icon" width="32" height="32">

      <a href="https://github.blog/changelog/2021-11-10-github-actions-input-types-for-manual-workflows/" target="_blank" rel="noopener">The GitHub Blog</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/362;"><img src="https://github.blog/wp-content/uploads/2022/04/Engineering-Security.png?fit=1200%2C630" class="thumbnail" width="690" height="362"></div>

<h3><a href="https://github.blog/changelog/2021-11-10-github-actions-input-types-for-manual-workflows/" target="_blank" rel="noopener">GitHub Actions: Input types for manual workflows | GitHub Changelog</a></h3>

  <p>GitHub Actions: Input types for manual workflows</p>

  <p>
    <span class="label1">Est. reading time: 1 minute</span>
  </p>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

**Comment by abel-rm-rf** [comment]
<p>Input works fine for manual workflow but I need something that will persist. i.e once the trigger branch has been changed I want it to stay like that for subsequent workflows triggered automatically.<br>
For e.g if I was to use secrets as a config it would look something like this:</p>
<pre><code class="lang-auto">name: Build &amp; Deploy
on:
  workflow_dispatch:
  push:
    branches:
      - ${{ secrets.BRANCH_TO_WATCH }} # I want to change this dynamically
....
</code></pre>

**Comment by jsoref** [comment, marked as answer]
<p>I’d probably delete the branches constraint and then have a step that runs if it’s a push and checks to see if the branch is master or a specific file exists and sets an output. And then have the rest of the workflow run only if that output is set.</p>
<p>Then you can have the input workflow create that file on that branch and push.</p>

**Comment by abel-rm-rf** [comment]
<p>Ahh that is exactly what I need. I added an if check in the build step and it works exactly as I need -<br>
Thank you so much!<br>
Here was my final solution for others that may come across this in the future.</p>
<pre><code class="lang-auto">name: Build &amp; Deploy
on:
  workflow_dispatch:
  push:
jobs:
  build:
    env: ${{secrets.BRANCH_TO_WATCH}}
    runs-on: ubuntu-latest
    steps:
      - name: install &amp; build
        if: ${{ contains(github.ref, env.BRANCH_TO_DEPLOY) }} # &lt;-----
        run: ....
</code></pre>
